### PR TITLE
Use Redis-based node discovery

### PR DESCRIPTION
**Pull request by parrish**
This PR switches from zookeeper to Redis for node discovery.  This also removes the Celluloid-backed client -- node set changes occur in threaded callbacks now.

It'd be good to set this up on staging and verify everything.

The discovery library is at https://github.com/parrish/attention

The version of cellect that uses is it is at https://github.com/zooniverse/cellect/tree/attention ([diff](https://github.com/zooniverse/Cellect/compare/attention))

### CODE REVIEW

**Comment by camallen** [comment]
This all looks good. 

We should probably modify the cellect start puma script to use a redis host / db key [here](https://github.com/zooniverse/Cellect/blob/9e89e052765f92c95f5c0bf2dc210cf6a23ff1b3/script/start_puma#L28). 

Also will the client redis connection lazy load (wondering about forking puma configs)?

File: cellect_env.rb [hunk 1]
```diff
@@ -12,9 +12,9 @@ def setup_env_vars
     setup_environment
     preload_workflows
     load_db_yaml
-    load_zk_yaml
+    load_redis_yaml
     @env_vars = {
-      "ZK_URL" => @zk_url,
+      "ATTENTION_REDIS_URL" => @redis_url,
```
Comment (camallen) [inline cellect_env.rb]: can we use the redis_url method here instead to allow an explicit ENV var to override any yaml / other env configs?

File: cellect_env.rb [hunk 2]
```diff
@@ -83,8 +83,8 @@ def database_url
     "postgresql://#{@pg_user}:#{@pg_pass}@#{@pg_host}:#{@pg_port}/#{@pg_db}?pool=#{connection_pool_value}"
   end
 
-  def zk_url
-    ENV['ZK_URL'] ||
-    "#{ENV["ZK_PORT_2181_TCP_ADDR"]}:#{ENV["ZK_PORT_2181_TCP_PORT"]}"
+  def redis_url
+    ENV['ATTENTION_REDIS_URL'] ||
+    "redis://#{ENV["REDIS_PORT_6379_TCP_ADDR"]}:#{ENV["REDIS_PORT_6379_TCP_PORT"]}/0"
```
Comment (camallen) [inline cellect_env.rb]: perhaps move the docker_compose ENV vars to a failed yaml lookup in the `load_redis_yaml` and then mimic the `database_url` method to use the yaml + fallback configs? this way we get an order of 1. explicit ENV var, 2. yaml config, 3. docker_compose defaults.

**Comment by parrish** [comment]
@camallen Mind checking this out again?

**Comment by camallen** [comment]
This looks good! I played around with the code when testing but may just issue another PR for those minor changes.

**Comment by parrish** [comment]
If we're good to go on this, I'll get zooniverse/Cellect#50 ready to go to master.

**Comment by camallen** [comment]
lets do it, i've turned off cellect in panoptes so we shouldn't impact that.

**Pull request merged**